### PR TITLE
Enforce at least one device group when creating a campaign.

### PR DESCRIPTION
**Pull request by houcros**

### CODE REVIEW

File: src/test/scala/com/advancedtelematic/campaigner/actor/CampaignSchedulerSpec.scala [hunk 1]
```diff
@@ -27,13 +28,13 @@ class CampaignSchedulerSpec extends ActorSpec[CampaignScheduler] with Campaigner
...
-    val groups   = arbitrary[Set[GroupId]].sample.get
-    val campaign = createDbCampaignWithUpdate(groups = groups).futureValue
+    val groups   = arbitrary[NonEmptyList[GroupId]].sample.get
+    val campaign = createDbCampaignWithUpdate(maybeGroups = Some(groups)).futureValue
 
     val parent   = TestProbe()
 
     campaigns.scheduleGroups(campaign.id, groups).futureValue
-    groups.foreach { g => deviceRegistry.setGroup(g, arbitrary[Seq[DeviceId]].sample.get) }
+    groups.map{ g => deviceRegistry.setGroup(g, arbitrary[Seq[DeviceId]].sample.get) }
```
Comment (simao) [inline src/test/scala/com/advancedtelematic/campaigner/actor/CampaignSchedulerSpec.scala]: why not keep foreach? foreach is usually used when the iteration is just for side effects

Comment (houcros) [reply src/test/scala/com/advancedtelematic/campaigner/actor/CampaignSchedulerSpec.scala]: There is no `foreach` for `NonEmptyList` 😕

**Comment by houcros** [comment]
@jerrytrieu do you think we're ready to merge this week? We haven't messed around with the campaigner for a while, should be stable by now?

**Comment by jerrytrieu** [comment]
@houcros let's merge

**Pull request merged**